// @primer/react/lib-esm/DropdownMenu2.js
function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

import React from 'react';
import { useSSRSafeId } from '@react-aria/ssr';
import { TriangleDownIcon } from '@primer/octicons-react';
import { Button } from './Button2';
import { AnchoredOverlay } from './AnchoredOverlay';
import { useProvidedRefOrCreate, useProvidedStateOrCreate, useMenuInitialFocus, useTypeaheadFocus } from './hooks';
import { Divider } from './ActionList2/Divider';
import { ActionListContainerContext } from './ActionList2/ActionListContainerContext';
const MenuContext = /*#__PURE__*/React.createContext({
  renderAnchor: null,
  open: false
});

const Menu = ({
  anchorRef: externalAnchorRef,
  open,
  onOpenChange,
  children
}) => {
  const [combinedOpenState, setCombinedOpenState] = useProvidedStateOrCreate(open, onOpenChange, false);
  const onOpen = React.useCallback(() => setCombinedOpenState(true), [setCombinedOpenState]);
  const onClose = React.useCallback(() => setCombinedOpenState(false), [setCombinedOpenState]);
  const anchorRef = useProvidedRefOrCreate(externalAnchorRef);
  const anchorId = useSSRSafeId();
  let renderAnchor = null; // ðŸš¨ Hack for good API!
  // we strip out Anchor from children and pass it to AnchoredOverlay to render
  // with additional props for accessibility

  const contents = React.Children.map(children, child => {
    if (child.type === MenuButton || child.type === Anchor) {
      renderAnchor = anchorProps => /*#__PURE__*/React.cloneElement(child, anchorProps);

      return null;
    }

    return child;
  });
  return /*#__PURE__*/React.createElement(MenuContext.Provider, {
    value: {
      anchorRef,
      renderAnchor,
      anchorId,
      open: combinedOpenState,
      onOpen,
      onClose
    }
  }, contents);
};

Menu.displayName = "Menu";
const Anchor = /*#__PURE__*/React.forwardRef(({
  children,
  ...anchorProps
}, anchorRef) => {
  return /*#__PURE__*/React.cloneElement(children, { ...anchorProps,
    ref: anchorRef
  });
});
/** this component is syntactical sugar ðŸ­ */

const MenuButton = /*#__PURE__*/React.forwardRef(({
  children,
  ...props
}, anchorRef) => {
  return /*#__PURE__*/React.createElement(Anchor, {
    ref: anchorRef
  }, /*#__PURE__*/React.createElement(Button, _extends({
    trailingIcon: TriangleDownIcon,
    type: "button"
  }, props), children));
});

const Overlay = ({
  children,
  ...overlayProps
}) => {
  // we typecast anchorRef as required instead of optional
  // because we know that we're setting it in context in Menu
  const {
    anchorRef,
    renderAnchor,
    anchorId,
    open,
    onOpen,
    onClose
  } = React.useContext(MenuContext);
  const containerRef = /*#__PURE__*/React.createRef();
  const {
    openWithFocus
  } = useMenuInitialFocus(open, onOpen, containerRef);
  useTypeaheadFocus(open, containerRef);
  return /*#__PURE__*/React.createElement(AnchoredOverlay, {
    anchorRef: anchorRef,
    renderAnchor: renderAnchor,
    anchorId: anchorId,
    open: open,
    onOpen: openWithFocus,
    onClose: onClose,
    overlayProps: overlayProps,
    focusZoneSettings: {
      focusOutBehavior: 'wrap'
    }
  }, /*#__PURE__*/React.createElement("div", {
    ref: containerRef
  }, /*#__PURE__*/React.createElement(ActionListContainerContext.Provider, {
    value: {
      container: 'DropdownMenu',
      listRole: 'menu',
      listLabelledBy: anchorId,
      selectionVariant: 'single',
      selectionAttribute: 'aria-checked',
      afterSelect: onClose
    }
  }, children)));
};

Overlay.displayName = "Overlay";
Menu.displayName = 'ActionMenu';
export const DropdownMenu = Object.assign(Menu, {
  Button: MenuButton,
  Anchor,
  Overlay,
  Divider
});
