// @primer/react/lib/Select.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _constants = require("./constants");

var _TextInputWrapper = _interopRequireDefault(require("./_TextInputWrapper"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

const StyledSelect = _styledComponents.default.select.withConfig({
  displayName: "Select__StyledSelect",
  componentId: "sc-li6bhs-0"
})(["appearance:none;background-color:transparent;border:0;color:currentColor;outline:none;width:100%;&:invalid{color:", ";}&:invalid option:not(:first-child){color:", ";}"], (0, _constants.get)('colors.fg.subtle'), (0, _constants.get)('colors.fg.default'));

const ArrowIndicatorSVG = ({
  className
}) => /*#__PURE__*/_react.default.createElement("svg", {
  width: "16",
  height: "16",
  fill: "currentColor",
  xmlns: "http://www.w3.org/2000/svg",
  className: className
}, /*#__PURE__*/_react.default.createElement("path", {
  d: "m4.074 9.427 3.396 3.396a.25.25 0 0 0 .354 0l3.396-3.396A.25.25 0 0 0 11.043 9H4.251a.25.25 0 0 0-.177.427ZM4.074 7.47 7.47 4.073a.25.25 0 0 1 .354 0L11.22 7.47a.25.25 0 0 1-.177.426H4.251a.25.25 0 0 1-.177-.426Z"
}));

ArrowIndicatorSVG.displayName = "ArrowIndicatorSVG";
const ArrowIndicator = (0, _styledComponents.default)(ArrowIndicatorSVG).withConfig({
  displayName: "Select__ArrowIndicator",
  componentId: "sc-li6bhs-1"
})(["pointer-events:none;position:absolute;right:4px;top:50%;transform:translateY(-50%);"]);

const Select = /*#__PURE__*/_react.default.forwardRef(({
  children,
  disabled,
  placeholder,
  size,
  required,
  validationStatus,
  ref: _propsRef,
  ...rest
}, ref) => /*#__PURE__*/_react.default.createElement(_TextInputWrapper.default, {
  sx: {
    position: 'relative'
  },
  size: size,
  validationStatus: validationStatus
}, /*#__PURE__*/_react.default.createElement(StyledSelect, _extends({
  ref: ref,
  required: required || Boolean(placeholder),
  disabled: disabled,
  "aria-required": required,
  "aria-disabled": disabled,
  "aria-invalid": validationStatus === 'error' ? 'true' : 'false'
}, rest), placeholder && /*#__PURE__*/_react.default.createElement("option", {
  value: "",
  disabled: required,
  selected: true,
  hidden: required
}, placeholder), children), /*#__PURE__*/_react.default.createElement(ArrowIndicator, null)));

const Option = props => /*#__PURE__*/_react.default.createElement("option", props);

Option.displayName = "Option";

const OptGroup = props => /*#__PURE__*/_react.default.createElement("optgroup", props);

OptGroup.displayName = "OptGroup";

var _default = Object.assign(Select, {
  Option,
  OptGroup
});

exports.default = _default;
